cmd/bpf2go/makedep_test.go

```diff
@@ -59,3 +60,34 @@ nothing:
 		t.Error("Output doesn't match")
 	}
 }
+
+func TestMergeDependencies(t *testing.T) {
+	deps1 := []dependency{
+		{"/foo/main.c", []string{"/foo/bar.h", "/foo/baz.h"}},
+		{"/foo/other.c", []string{"/foo/bar.h"}},
+	}
+
+	deps2 := []dependency{
+		{"/foo/main.c", []string{"/foo/qux.h"}},
+		{"/foo/third.c", []string{"/foo/bar.h"}},
+	}
+
+	merged := mergeDependencies(deps1, deps2)
+
+	// Sort merged dependencies for stable comparison
+	sort.Slice(merged, func(i, j int) bool {
+		return merged[i].file < merged[j].file
+	})
+
+	want := []dependency{
+		{"/foo/main.c", []string{"/foo/bar.h", "/foo/baz.h", "/foo/qux.h"}},
```

Comment:
How can we get different dependencies for two compilations of the same file? Wouldn't it be more appropriate to check that both sets of dependencies are identical? Or that the sets are disjoint?